### DISCUSSION TITLE
Virtual/non-existing fields

### DISCUSSION THREAD

**Discussion by pjaspers** [jeremyevans/forme#131] [General]
I have a sequel model, that has a finished_at and a dnf_at column, now in my form I would like to have 3 radio buttons: not finished, finished or dnf (it's one of the three, they're mutually exclusive), and when submitting I'll set a datetime in the right column.

What would be the best way to handle this in forme? I I add a `finished` attribute, forme will generate a text input, I can probably hack around it but it feels like I'm missing something.

**Comment by jeremyevans** [comment]
You can use the `:type` option to set the type for a virtual field.  My guess is you would want `type: :radioset`, with the appropriate `:options` and `:value` (and likely other options as well).

**Reply by pjaspers** [reply]
Thanks, I wound up doing pretty much that (but the radioset option might be cleaner)

```
        <% [
          ["Not Finished", "not_finished", !f.obj.finished?],
          ["Finished", Route::FINISHED, !!f.obj.finished_at],
          ["DNF-ed", Route::DNF, !!f.obj.dnf_at],
        ].each do |(text, value, checked)| %>
          <%= f.input(:finish_status,
                      type: :radio,
                      label: text,
                      value: value,
                      label_attr: {class: "form-control form-control--radio"},
                      checked: checked,
          ) %>
        <% end %>
```
 and then handling the logic part inside the model with:

```
  def finish_status=(value)
    self.finished_at = nil
    self.dnf_at = nil
    case value
    when FINISHED
      self.finished_at = Time.now
    when DNF
      self.dnf_at = Time.now
    end
  end
  ```
